cmake/modules/FindCrossGUID.cmake

```diff
@@ -1,43 +1,85 @@
+# FindCrossGUID
+# -------
+# Finds the CrossGUID library
+#
+# This will define the following variables::
+#
+# CROSSGUID_FOUND_FOUND - system has CrossGUID
+# CROSSGUID_INCLUDE_DIRS - the CrossGUID include directory
+# CROSSGUID_LIBRARIES - the CrossGUID libraries
+# CROSSGUID_DEFINITIONS - cmake definitions required
+#
+# and the following imported targets::
+#
+#   crossguid   - The CrossGUID library
+
 if(ENABLE_INTERNAL_CROSSGUID)
   include(cmake/scripts/common/ModuleHelpers.cmake)
 
   set(MODULE_LC crossguid)
 
+  # Temp: We force CMAKE_BUILD_TYPE to release, and makefile builds respect this
+  # Multi config generators (eg VS, Xcode) dont, so handle debug postfix build/link for them only
+  if(NOT CMAKE_GENERATOR STREQUAL "Unix Makefiles")
```

Comment:
This probably isn't quite right as the ninja generator should be considered similar to Unix makefiles.

Reply:
added a check for ninja generator in #21178